src/server.js

```diff
@@ -1,45 +1,50 @@
+const _ = require('lodash/fp')
 const express = require('express')
 const cors = require('cors')
 const bodyParser = require('body-parser')
 const { promiseHandler, Response, validateInput } = require('./utils')
-const { project, samRoot } = require('../config')
+const { project, orchestrationRoot, samRoot } = require('../config')
 const { logger, getSecret } = require('./google-utils')
 const btoa = require('btoa-lite')
 const fetch = require('node-fetch')
 const Joi = require('@hapi/joi')
 
-
-const withAuth = wrappedFn => async (req, ...args) => {
-  const authRes = await fetch(
-    `${samRoot}/register/user/v2/self/info`,
-    { headers: { authorization: req.headers.authorization } }
-  ).catch(() => {
-    throw new Response(503, 'Unable to contact auth service')
+const fetchOk = async (url, { serviceName, ...options } = {}) => {
+  const res = await fetch(url, options).catch(() => {
+    throw new Response(503, `Unable to contact ${serviceName} service`)
   })
-
-  if (authRes.status === 401) {
+  if (res.status === 401) {
     throw new Response(401, 'Unauthorized')
   }
-
-  if (!authRes.ok) {
-    console.error('Sam error', await authRes.text())
-    throw new Response(503, 'Failed to query auth service')
+  if (!res.ok) {
+    console.error(`${serviceName} error`, await res.text())
+    throw new Response(503, `Failed to query ${serviceName} service`)
   }
+  return res
+}
 
-  const { enabled } = await authRes.json()
-  if (!enabled) {
-    throw new Response(403, 'Forbidden')
+const fetchMixpanel = async (url, { data, ...options } = {}) => {
+  const res = await fetchOk(
+    `https://api.mixpanel.com/${url}?data=${btoa(JSON.stringify(data))}`,
+    { serviceName: 'metrics', ...options }
```

Comment:
How is serviceName being used in the call?

Reply:
It's just descriptive text used in error messages (see `fetchOk` above)